src/clients/SpokePoolClient.ts

```diff
@@ -297,8 +298,7 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @returns The corresponding deposit if found, undefined otherwise.
    */
   public getDeposit(depositId: BigNumber): DepositWithBlock | undefined {
-    const depositHash = this.getDepositHash({ depositId, originChainId: this.chainId });
-    return this.depositHashes[depositHash];
+    return Object.values(this.depositHashes).find(({ depositId: _depositId }) => _depositId.eq(depositId));
```

Comment:
For `getDeposit()` we have to revert to a search because the only input available is the depositId. This is not robust to duplicate depositIds.